pageserver/src/tenant/secondary.rs

```diff
@@ -110,6 +110,27 @@ impl SecondaryTenant {
     fn get_tenant_shard_id(&self) -> &TenantShardId {
         &self.tenant_shard_id
     }
+
+    /// For API access: generate a LocationConfig equivalent to the one that would be used to
+    /// create a Tenant in the same state.  Do not use this in hot paths: it's for relatively
+    /// rare external API calls, like a reconciliation at startup.
+    pub(crate) fn get_location_conf(&self) -> models::LocationConfig {
+        let conf = self.detail.lock().unwrap().config.clone();
+
+        let conf = models::LocationConfigSecondary { warm: conf.warm };
+
+        models::LocationConfig {
+            mode: models::LocationConfigMode::Secondary,
+            generation: None,
+            secondary_conf: Some(conf),
+            shard_number: self.tenant_shard_id.shard_number.0,
+            shard_count: self.tenant_shard_id.shard_count.0,
+            // Shard stripe size is not set in secondary mode
+            shard_stripe_size: u32::default(),
```

Comment:
I wonder if `0` would be better here: iirc that value is not a legal stripe size.

Reply:
u32::default() is zero -- this is used for symmetry with the `#[serde(default)]` on the structure.

Reply:
yeah I'm aware, the `default` hides it. But I'm okay with `default` as well.